app/http/httpclient.c

```diff
@@ -218,297 +219,297 @@ static int ICACHE_FLASH_ATTR http_chunked_decode( const char * chunked, char * d
 	}
 
-	if(req->headers == NULL) /* Avoid NULL pointer, it may cause exception */
-	{
+	if (req->headers == NULL) { /* Avoid NULL pointer, it may cause exception */
 		req->headers = (char *)os_malloc(sizeof(char));
 		req->headers[0] = '\0';
 	}
 
-	char buf[69 + strlen( req->method ) + strlen( req->path ) + strlen( req->hostname ) +
-		 strlen( req->headers ) + strlen( post_headers )];
-	int len = os_sprintf( buf,
-			      "%s %s HTTP/1.1\r\n"
-			      "Host: %s:%d\r\n"
-			      "Connection: close\r\n"
-			      "User-Agent: ESP8266\r\n"
-			      "%s"
-			      "%s"
-			      "\r\n",
-			      req->method, req->path, req->hostname, req->port, req->headers, post_headers );
-
-	if ( req->secure )
-		espconn_secure_send( conn, (uint8_t *) buf, len );
+	char buf[69 + strlen(req->method) + strlen(req->path) + strlen(req->hostname) +
+	         strlen(req->headers) + strlen(post_headers)];
+	int len = os_sprintf(buf,
+	                     "%s %s HTTP/1.1\r\n"
+	                     "Host: %s:%d\r\n"
+	                     "Connection: close\r\n"
+	                     "User-Agent: ESP8266\r\n"
+	                     "%s"
+	                     "%s"
+	                     "\r\n",
+	                     req->method, req->path, req->hostname, req->port, req->headers, post_headers);
+
+	if (req->secure)
+		espconn_secure_send(conn, (uint8_t *) buf, len);
 	else
-		espconn_send( conn, (uint8_t *) buf, len );
-	if(req->headers != NULL)
-		os_free( req->headers );
+		espconn_send(conn, (uint8_t *) buf, len);
+
+	if (req->headers != NULL)
+		os_free(req->headers);
+
 	req->headers = NULL;
-	HTTPCLIENT_DEBUG( "Sending request header\n" );
+	HTTPCLIENT_DEBUG("Sending request header\n");
 }
 
 
-static void ICACHE_FLASH_ATTR http_disconnect_callback( void * arg )
+static void ICACHE_FLASH_ATTR http_disconnect_callback(void * arg)
 {
-	HTTPCLIENT_DEBUG( "Disconnected\n" );
+	HTTPCLIENT_DEBUG("Disconnected\n");
 	struct espconn *conn = (struct espconn *) arg;
 
-	if ( conn == NULL )
-	{
+	if (conn == NULL) {
 		return;
 	}
 
-	if ( conn->proto.tcp != NULL )
-	{
-		os_free( conn->proto.tcp );
+	if (conn->proto.tcp != NULL) {
+		os_free(conn->proto.tcp);
 	}
-	if ( conn->reverse != NULL )
-	{
-		request_args_t	* req		= (request_args_t *) conn->reverse;
-		int		http_status	= -1;
-		char		* body		= "";
-
-		// Turn off timeout timer
-		os_timer_disarm( &(req->timeout_timer) );
-
-		if ( req->buffer == NULL )
-		{
-			HTTPCLIENT_DEBUG( "Buffer shouldn't be NULL\n" );
-		}
-		else if ( req->buffer[0] != '\0' )
-		{
+
+	if (conn->reverse != NULL) {
+		request_args_t  * req           = (request_args_t *) conn->reverse;
+		int             http_status     = -1;
+		char            * body          = "";
+
+		/* Turn off timeout timer */
+		os_timer_disarm(&(req->timeout_timer));
+
+		if (req->buffer == NULL) {
+			HTTPCLIENT_DEBUG("Buffer shouldn't be NULL\n");
+			/* To avoid NULL buffer */
+			req->buffer = "";
```

Comment:
I'm not saying that changing null to an empty string is wrong, but calling the C runtime 'fee' on a statically allocated string is WRONG (often fatally wrong). This is heap corruption.

You could allocate one byte and set it to zero and make a dynamically allocated empty string. This could be freed later. There may be a better approach.